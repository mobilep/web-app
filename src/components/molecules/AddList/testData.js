export const users = [
	{
		id: 1,
		fullName: 'John Smith',
		initials: 'JS',
	}, {
		id: 2,
		fullName: 'Philippe Starck',
		initials: 'PS',
	}, {
		id: 3,
		fullName: 'Suzy Salmon',
		initials: 'SS',
	},
];

export const allUsers = [
	{
		id: 4,
		fullName: 'John Rebele',
		initials: 'JR',
	}, {
		id: 5,
		fullName: 'John Klokdan',
		initials: 'JK',
	}, {
		id: 6,
		fullName: 'John Makeba',
		initials: 'JM',
	},
];
