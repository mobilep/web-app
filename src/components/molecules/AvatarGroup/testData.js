export const avatars = [
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
];

export const avatarsLess = [
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
];
