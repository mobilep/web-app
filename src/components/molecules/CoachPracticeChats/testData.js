/* eslint-disable max-len */
export const practiceChats = [
	{
		currentUserId: '5d6e509093b6e912b83a65ce',
		_id: '5dad8ef5d3c420b10c618100',
		_coach: {
			_id: '5d6e509093b6e912b83a65ce',
			firstName: 'Roman',
			lastName: 'Yavoriv',
			avatarId: '5f549515-ee86-40f2-9104-5c6a8a80beb7',
			devices: 'Array[0]',
			avatarColor: '58c9ef',
			lang: 'en',
			avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_md: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/640/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_lg: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/1024/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			havePassword: false,
			name: 'Roman Yavoriv',
			extraInformation: 'Array[7]',
		},
		_scenario: {
			_id: '5dad8ef546396073dd363907',
			groupChat: 'https://mobile-practice-dev.firebaseio.com/mpchat/5dad8ef546396073dd363909',
			isPracticed: true,
			isActive: true,
			name: '2nd scenario',
			video: 'Object',
		},
		_user: {
			_id: '5d70bc6d93b6e912b83a65d5',
			firstName: 'Taras',
			lastName: 'Kvas',
			avatarId: '2e3db607-9ec1-45f4-a7e5-4bed03912dd5',
			devices: 'Array[0]',
			avatarColor: 'fd524f',
			lang: 'en',
			avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
			avatar_md: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/640/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
			avatar_lg: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/1024/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
			havePassword: false,
			name: 'Taras Kvas',
			extraInformation: 'Array[7]',
		},
		__v: 0,
		createdAt: '2019-10-21T10:56:53.554Z',
		updatedAt: '2019-10-29T09:19:42.764Z',
		userEvaluatedAt: null,
		coachEvaluatedAt: null,
		hasBestPractice: false,
		hasUserMessage: true,
		isActive: true,
		lastMessageFrom: 'learner',
		status: 'current',
		coachMark: [],
		userMark: [],
		coachAudiosCount: 0,
		coachPhotosCount: 0,
		coachFilesCount: 0,
		coachVideosCount: 0,
		coachMessagesCount: 4,
		userAudiosCount: 0,
		userPhotosCount: 1,
		userFilesCount: 0,
		userVideosCount: 0,
		userMessagesCount: 19,
		_inbox: {
			_id: '5dad8ef546396073dd36390c',
			updatedAt: '2019-10-29T09:20:28.648Z',
			createdAt: '2019-10-21T10:56:53.544Z',
			_user: '5d6e509093b6e912b83a65ce',
			_recipient: '5d70bc6d93b6e912b83a65d5',
			__v: 0,
			inboxId: 'chats/5d6e509093b6e912b83a65ce/5dad8ef546396073dd363907/5dad8ef5d3c420b10c618100/5dad8ef546396073dd36390c',
			lastReadMessage: '-LsLo4OhHveQpLP5JXKy',
			unreadMessagesCount: 0,
			type: 'practice',
			status: 'read',
			message: 'Object',
			firstResponseAt: '2019-10-28T13:25:34.142Z',
			firstMessageAt: '2019-10-28T13:09:17.159Z',
			isActive: true,
		},
		state: 'waitingOnFeedback',
		messages: [
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
		],
	},
	{
		currentUserId: '5d6e509093b6e912b83a65ce',
		_id: '5dad8ef5d3c420b10c618102',
		_coach: {
			_id: '5d6e509093b6e912b83a65ce',
			firstName: 'Roman',
			lastName: 'Yavoriv',
			avatarId: '5f549515-ee86-40f2-9104-5c6a8a80beb7',
			devices: 'Array[0]',
			avatarColor: '58c9ef',
			lang: 'en',
			avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_md: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/640/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			avatar_lg: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/1024/5f549515-ee86-40f2-9104-5c6a8a80beb7.jpg',
			havePassword: false,
			name: 'Roman Yavoriv',
			extraInformation: 'Array[7]',
		},
		_scenario: {
			_id: '5dad8ef546396073dd363907',
			groupChat: 'https://mobile-practice-dev.firebaseio.com/mpchat/5dad8ef546396073dd363909',
			isPracticed: true,
			isActive: true,
			name: '2nd scenario',
			video: 'Object',
		},
		_user: {
			_id: '5da86a3716973b29347321ce',
			firstName: 'Roman',
			lastName: 'Yavoriv',
			devices: 'Array[0]',
			avatarColor: '00cdbb',
			lang: 'en',
			avatar_sm: null,
			avatar_md: null,
			avatar_lg: null,
			havePassword: false,
			name: 'Roman Yavoriv',
			extraInformation: 'Array[7]',
		},
		__v: 0,
		createdAt: '2019-10-21T10:56:53.563Z',
		updatedAt: '2019-10-29T09:18:36.340Z',
		userEvaluatedAt: '2019-10-29T09:16:14.949Z',
		coachEvaluatedAt: null,
		hasBestPractice: false,
		hasUserMessage: false,
		isActive: true,
		lastMessageFrom: 'coach',
		coachMark: [
			'Object',
		],
		userMark: [],
		coachAudiosCount: 0,
		coachPhotosCount: 0,
		coachFilesCount: 0,
		coachVideosCount: 0,
		coachMessagesCount: 2,
		userAudiosCount: 0,
		userPhotosCount: 0,
		userFilesCount: 0,
		userVideosCount: 0,
		userMessagesCount: 0,
		_inbox: {
			_id: '5dad8ef546396073dd36390d',
			updatedAt: '2019-10-29T09:19:11.848Z',
			createdAt: '2019-10-21T10:56:53.547Z',
			_user: '5d6e509093b6e912b83a65ce',
			_recipient: '5da86a3716973b29347321ce',
			__v: 0,
			inboxId: 'chats/5d6e509093b6e912b83a65ce/5dad8ef546396073dd363907/5dad8ef5d3c420b10c618102/5dad8ef546396073dd36390d',
			lastReadMessage: '-LsLnpAfH4s1IOZsQrgY',
			unreadMessagesCount: 0,
			type: 'practice',
			status: 'read',
			message: 'Object',
			firstResponseAt: null,
			firstMessageAt: '2019-10-29T09:18:27.588Z',
			isActive: true,
		},
		state: 'completed',
		messages: [
			'Object',
			'Object',
			'Object',
			'Object',
			'Object',
		],
	},
];

export const activeChatId = practiceChats[0]._id;
