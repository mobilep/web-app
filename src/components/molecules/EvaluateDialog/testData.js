export const evaluationList = [
	{
		id: '1',
		title: 'Contactâ€™s attention grabbed',
	}, {
		id: '2',
		title: 'Self Introduction',
	}, {
		id: '3',
		title: 'Purpose of call stated',
	},
];
