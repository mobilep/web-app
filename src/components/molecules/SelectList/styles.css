.SelectList {
	width: 100%;
	padding: 4px 0;
	background-color: var(--white);
	box-shadow: 0 2px 4px 0 var(--transparentGrey1);
}

.SelectList-item {
	display: flex;
	width: 100%;
	align-items: center;
	min-height: 32px;
	padding: 0 12px;
	margin: 0;
	background-color: var(--white);
	border: none;
	will-change: background-color;
	transition: background-color .2s ease-in-out;
	cursor: pointer;
}

.SelectList-item-active {
	background-color: var(--grey2);
}

.SelectList-item-customItemWrapper {
	border-top: 1px solid var(--grey2);
	padding-top: 8px;
	padding-bottom: 4px;
}
