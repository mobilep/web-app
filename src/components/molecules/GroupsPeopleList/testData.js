import { avatars } from '../AvatarGroup/testData';

export const groups = [
	{
		_id: '5b3b84595d7f59044c6759ea',
		name: 'Techmagic team',
		_users: avatars,
	},
	{
		_id: '5b3b84595d7f59044c6759eb',
		name: 'SoftServe team',
		_users: avatars,
	},
	{
		_id: '5b3b84595d7f59044c6759ec',
		name: 'Epam team',
		_users: avatars,
	},
];
