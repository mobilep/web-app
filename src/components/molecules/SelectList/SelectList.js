import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './styles.css';

export default class SelectList extends Component {

	static propTypes = {
		className: PropTypes.string,
		itemIdKey: PropTypes.string,
		itemValueKey: PropTypes.string,
		items: PropTypes.array.isRequired,
		onMouseOver: PropTypes.func,
		onSelect: PropTypes.func,
		selectedItem: PropTypes.number,
	}

	static defaultProps = {
		itemIdKey: 'id',
		itemValueKey: 'value',
		onMouseOver: () => {},
		onSelect: () => {},
	}

	getClassName (className) {
		return classNames('SelectList', className);
	}

	getItemClassName (isActive) {
		return classNames({
			'SelectList-item': true,
			'SelectList-item-active': isActive,
		});
	}

	handleClick = (id) =>
		() => this.props.onSelect(id)

	handleMouseOver = (index) =>
		() => this.props.onMouseOver(index);

	getItem (element, isCustom) {
		return isCustom
			? (
				<div className='SelectList-item-customItemWrapper' key={'isCustom'}>
					{element}
				</div>
			)
			: element;
	}

	getItems ({ items, selectedItem, itemIdKey, itemValueKey } = this.props) {
		return items.map((item, index) => {
			const id = item[itemIdKey];
			const value = item[itemValueKey];
			let { description = '' } = item;
			const { isCustom } = item;
			description = description && ` ${description}`;
			const key = `selectItem${index}`;
			const isActive = selectedItem === index;
			const className = this.getItemClassName(isActive);
			return this.getItem(
				<button
					className={className}
					key={key}
					onMouseOver={this.handleMouseOver(index)}
					onMouseLeave={this.handleMouseOver(null)}
					onClick={this.handleClick(id)}
				>
					{value}{description}
				</button>,
				isCustom
			);
		});
	}

	render () {
		const { className } = this.props;
		return (
			<div className={this.getClassName(className)}>
				{this.getItems()}
			</div>
		);
	}

}
