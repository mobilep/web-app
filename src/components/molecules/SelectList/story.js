import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import SelectList from './SelectList';
import { items } from './testData';

storiesOf('2. Molecules / SelectList', module)
	.add('SelectList', () => (
		<SelectList items={items} selectedItem={1} onSelect={action('onSelect')} />
	));
