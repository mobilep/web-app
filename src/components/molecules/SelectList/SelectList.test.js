import React from 'react';
import { shallow } from 'enzyme';

import SelectList from './SelectList';

import { items } from './testData';

describe('Molecule - SelectList', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<SelectList selectedItem={1} items={items} />
		);
		expect(wrapper).toMatchSnapshot();
	});

});
