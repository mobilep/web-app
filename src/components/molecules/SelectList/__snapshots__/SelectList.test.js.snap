// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - SelectList Should render 1`] = `
<div
  className="SelectList"
>
  <button
    className="SelectList-item"
    key="selectItem0"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    first item
  </button>
  <button
    className="SelectList-item SelectList-item-active"
    key="selectItem1"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    second item
  </button>
  <button
    className="SelectList-item"
    key="selectItem2"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    third item
  </button>
</div>
`;
